docs/database/schema.sql

```diff
@@ -103,3 +103,28 @@ CREATE TABLE action_instance
     CONSTRAINT action_instance_configured_connector_id_fk FOREIGN KEY (configured_connector_id) REFERENCES configured_connector (id)
 );
 
+-- An integration is the integration definition. It contains a list of integration steps.
+CREATE TABLE integration
+(
+    id SERIAL PRIMARY KEY NOT NULL,
+    name VARCHAR(256) NOT NULL,
+    description VARCHAR(2048)
+);
+CREATE UNIQUE INDEX integration_name_uindex ON integration (name);
+
+-- An integration step is the definition of a single step in an integration. It contains configuration
+-- for each and every step in the integration flow.
+-- This can reference an action instance or be a supported integration pattern (e.g. log, wiretap, choice, etc).
+CREATE TABLE integration_step
+(
+    id SERIAL PRIMARY KEY NOT NULL,
+    integration_id INTEGER NOT NULL,
+    action_instance_id INTEGER,
+    step_index SMALLINT NOT NULL,
+    step_type VARCHAR(32) NOT NULL,
```

Comment:
Yeah that's right, this is a discriminator, but in Java world it doesn't really have any need for inheritance hierarchy (I don't think) as they're simple steps. There could be an argument in future for step validation I guess where that kind of discriminator could be useful. This is more for the UI to identify what to display and also when we're creating the integration how to marshal this to a flow definition.